plugins/InspectionGadgets/src/com/intellij/codeInspection/RedundantCompareToJavaTimeInspection.java

```diff
@@ -0,0 +1,152 @@
+// Copyright 2000-2022 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
+package com.intellij.codeInspection;
+
+import com.intellij.codeInspection.dataFlow.DfaPsiUtil;
+import com.intellij.codeInspection.dataFlow.value.RelationType;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.util.NlsSafe;
+import com.intellij.psi.*;
+import com.intellij.psi.util.PsiTreeUtil;
+import com.intellij.psi.util.PsiUtil;
+import com.intellij.util.ObjectUtils;
+import com.siyeh.InspectionGadgetsBundle;
+import com.siyeh.ig.callMatcher.CallMatcher;
+import com.siyeh.ig.psiutils.CommentTracker;
+import com.siyeh.ig.psiutils.ExpressionUtils;
+import org.jetbrains.annotations.NotNull;
+
+public class RedundantCompareToJavaTimeInspection extends AbstractBaseJavaLocalInspectionTool implements CleanupLocalInspectionTool {
+
+  private static final String JAVA_TIME_LOCAL_DATE = "java.time.LocalDate";
+  private static final String JAVA_TIME_LOCAL_DATE_TIME = "java.time.LocalDateTime";
+  private static final String JAVA_TIME_LOCAL_TIME = "java.time.LocalTime";
+  private static final String JAVA_TIME_OFFSET_DATE_TIME = "java.time.OffsetDateTime";
+  private static final String JAVA_TIME_OFFSET_TIME = "java.time.OffsetTime";
+
+  private static final CallMatcher COMPARE_TO_METHODS = CallMatcher.anyOf(
+    CallMatcher.instanceCall(JAVA_TIME_LOCAL_DATE, "compareTo").parameterTypes("java.time.chrono.ChronoLocalDate"),
+    CallMatcher.instanceCall(JAVA_TIME_LOCAL_TIME, "compareTo").parameterTypes(JAVA_TIME_LOCAL_TIME),
+    CallMatcher.instanceCall(JAVA_TIME_LOCAL_DATE_TIME, "compareTo")
+      .parameterTypes("java.time.chrono.ChronoLocalDateTime"),
+    CallMatcher.instanceCall(JAVA_TIME_OFFSET_TIME, "compareTo").parameterTypes(JAVA_TIME_OFFSET_TIME),
+    CallMatcher.instanceCall(JAVA_TIME_OFFSET_DATE_TIME, "compareTo")
+      .parameterTypes(JAVA_TIME_OFFSET_DATE_TIME)
+  );
+
+  @Override
+  public @NotNull String getID() {
```

Comment:
The default ID is the `shortName`, so it is not necessary to override `getID()` if the ID is the same as the shortName.

Reply:
Deleted, Thank you!